Top-Level Requirements.md

```diff
@@ -1,63 +1,86 @@
+* 7.1.3 The navigation subsystem **shall** move the rover towards the goal, based on path information.
+* 7.1.4 The navigation subsystem **shall** cease operation upon arriving at a goal within a set tolerance.
+* 7.1.5 The navigation subsystem **shall**  provide appropriate error handling procedures.
+#### 7.2  Pre-Mining
+* 7.2.1 The software **shall** define a point at a configurable distance directly East of the bin , and within the mining zone as the initial goal.
+* 7.2.2 On navigating to the bin the subsystem **shall** optimize the current goal to minimize driving time for the rover.
+* 7.2.3 The algorithm **shall** define a line segment that represents the set of “minable” locations using the points L0, L1, and L2, where:
+    * 7.2.3.1 The point L1 is defined as a point directly East of the bin X meters, where X is the contest requirement for mining distance plus some configurable safety threshold.
+    * 7.2.3.2 The point L0 is defined as a point directly East of the robot and directly North or South of L1.
+    * 7.2.3.3 The point L2 is defined as a point flipped from L0 over L1.
+* 7.2.3 The software **shall** have a configurable time value GOAL_RECALC_DELTA that represents the time between recalculations
+* 7.2.4 The software **shall** calculate the nearest point on the the Mining Line every GOAL_RECALC_DELTA time units and use that nearest point as the navigation goal
+#### 7.3  Post-Mining
+* 7.3.1 The software **shall** define a static point at a configurable distance directly East of the bin as the goal and navigate there.
+### 8.  Mining Subsystem
+* 8.1 The mining subsystem **shall** be provided a time limit and not operate longer than that time limit.
+* 8.2 The mining subsystem **shall** hold a predefined mining procedure.
+* 8.3 The mining subsystem **shall** follow a predefined series of mining movements during the mining procedure.
+* 8.4 The mining procedure **shall** define an ordered set of mining locations (and dumping locations) relative to the robot position
+* 8.5 The mining procedure **shall** further be divided into atomically-operating subprocedures (scoops)
+* 8.6 The mining procedure **shall** always check timing flags before beginning scoops
+* 8.7 The mining procedure **shall** be terminated upon detecting timing flags
+* 8.8 Upon completing the mining procedure, the mining subsystem **shall**:
+    * 8.8.1 Define an “obstacle” representing the digging hole and send it to the navigation subsystem.
+    * 8.8.2 Return the mining arm to the starting position.
+### 9.  Dumping Subsystem
+* 9.1 As a precondition the rover **shall** be facing east, relatively close to the dumping receptacle.
+* 9.2 The dumping subsystem **shall** perform the localization procedure prior to the start of the  dumping procedure.
+* 9.3 The dumping procedure **shall**:
+    * 9.3.1 Orient itself and turn to face east.
+    * 9.3.2 Back up to within a set tolerance of the wall, maintaining eastern orientation.
```

Comment:
9.3.3 Move the digging arm out of the bin to the predefined dumping position.

Reply:
(this is my proposed addition here, feel free to change it to a way that's better phrased)